great_expectations/rule_based_profiler/expectation_configuration_builder/default_expectation_configuration_builder.py

```diff
@@ -37,30 +77,141 @@ def __init__(
         if not isinstance(meta, dict):
             raise ge_exceptions.ProfilerExecutionError(
                 message=f"""Argument "{meta}" in "{self.__class__.__name__}" must be of type "dictionary" \
-(value of type "{str(type())}" was encountered).
+(value of type "{str(type(meta))}" was encountered).
 """
             )
 
+        if condition and (not isinstance(condition, str)):
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"""Argument "{condition}" in "{self.__class__.__name__}" must be of type "string" \
+(value of type "{str(type(condition))}" was encountered).
+"""
+            )
+
+        self._condition = condition
         self._meta = meta
 
     @property
     def expectation_type(self) -> str:
         return self._expectation_type
 
     @property
-    def kwargs(self) -> dict:
-        return self._kwargs
+    def condition(self) -> str:
+        return self._condition
 
     @property
     def meta(self) -> dict:
         return self._meta
 
+    @property
+    def kwargs(self) -> dict:
+        return self._kwargs
+
+    def _parse_condition(self) -> ParseResults:
+        """
+        Using the grammer defined by expr, provides the condition to the parser.
+
+        Applicability: To be used as part of configuration (e.g., YAML-based files or text strings).
+        Extendability: Readily extensible to include "slice" and other standard accessors (as long as no dynamic elements).
+        """
+
+        try:
+            return expr.parseString(self._condition)
+        except ParseException:
+            raise ExpectationConfigurationConditionParserError(
+                f'Unable to parse Expectation Configuration Condition: "{self._condition}".'
+            )
+
+    def _substitute_parameters_and_variables(
+        self,
+        term_list: Union[str, ParseResults],
+        domain: Domain,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ) -> ParseResults:
+        """Recursively substitute all parameters and variables in term list"""
+        token: Union[str, ParseResults]
+        for i, token in enumerate(term_list):
+            if isinstance(token, str) and token.startswith("$"):
```

Comment:
@NathanFarmer Are you sure that this `if` statement is needed?  I am guessing that the entire logic can be handled by `get_parameter_value_and_validate_return_type()`.  Happy to discuss (I may be missing a subtlety here).  Thanks!

Reply:
@alexsherstinsky as this method recurses through each term, it may encounter a parameter or variable provided in the condition, or (in most cases) it will encounter a list of terms, or an unrelated string such as an operator (e.g. ">"). Please let me know if you would like to discuss further.